vignettes/developers.Rmd

```diff
@@ -0,0 +1,358 @@
+glue(tr_("See <{url}> to learn more"))
+```
+
+Similarly, if you're generating strings that include in HTML, avoid including the HTML in the translated string, and instead translate just the words:
+
+```{r, results = FALSE}
+# Instead of this:
+tr_("<a href='/index.html'>Home page</a>")
+
+# Try this:
+paste0("<a href='/index.html'>", tr_("Home page"), "</a>")
+```
+
+Generally, you want to help the translator spend as much time as possible helping you out.
+
+### Plurals
+
+In English, most nouns have different forms for one item (the singular) or more than one item (the plural, also used for zero items).
+So you might be tempted to construct a sentence like this:
+
+```{r}
+cows <- function(n) {
+  if (n == 1) {
+    paste0(n, " cow") 
+  } else { 
+    paste0(n, " cows")
+  }
+}
+paste("I have ", cows(0))
+paste("I have ", cows(1))
+paste("I have ", cows(2))
+```
+
+But this doesn't always work, even in English:
+
+```{r}
+paste0("There are ", cows(0), " in the field")
+paste0("There are ", cows(1), " in the field")
+```
+
+Again, we always want to construct a complete sentence:
+
+```{r}
+field_cows <- function(n) {
+  if (n == 1) {
+    fmt <- tr_("There is {n} cow in the field")
+  } else {
+    fmt <- tr_("There are {n} cows in the field")
+  }
+  glue(fmt)
+}
+```
+
+But there's an additional wrinkle here: while English has singular and plural, other languages have different forms.
+So we need to use a different helper: `ngettext(n, singular, plural)`:
+
+```{r}
+field_cows <- function(n) {
+  glue(ngettxt(n,
+    "There is {n} cow in the field",
+    "There are {n} cows in the field"
+  ))
+}
+```
+
+`ngettext()` generates a special form in the `.po` file, which shows both singular and plural forms:
+
+    msgid "There is {n} cow in the field"
+    msgid_plural "There are {n} cows in the field"
+
+Then the translator can supply any number of translations.
+Languages that don't have plurals (e.g. Chinese) only need to supply a single translation:
+
+    msgid "There is {n} cow in the field"
+    msgid_plural "There are {n} cows in the field"
+    msgstr[0] "田裡有{n}頭牛"
+
+Russian has three forms, so it gets three entries (roughly 1, 2-4, and everything else):
+
+    msgid "There is {n} cow in the field"
+    msgid_plural "There are {n} cows in the field"
+    msgstr[0] "В поле {xn} корова"
+    msgstr[1] "В поле {n} коровы"
+    msgstr[2] "В поле {n} коров"
+
+Slovenian and Serbian have four forms, Irish has five forms (learn more at [bitesize Irish](https://www.bitesize.irish/blog/counting/)), and Arabic has six forms.
+The rules that define which number get which message are quite complex and encoded in the "plural form" that's recorded at the top of the `.po` file and looks something like `(n%10==1 && n%100!=11 ? 0 : n%10>=2 && n%10<=4 && (n%100<10 || n%100>=20)? 1 : 2).` However, this is something the translators need to worry about not you, and as speakers of the language they should find it easier to puzzle out the rules.
```

Comment:
`potools` should also make it easier for them to puzzle out the rules!

Hence the `range` field here:

https://github.com/MichaelChirico/potools/blob/master/inst/extdata/plurals_metadata.csv

Reply:
I'd suggest exposing this as a standalone function. Maybe something like `po_explain_plural()`?